README.rst

```diff
@@ -211,14 +211,50 @@ actually accepts. Read more about it in the `HTML 5 Rocks CORS tutorial
 If ``True``, cookies will be allowed to be included in cross-site HTTP
 requests. Defaults to ``False``.
 
+``CORS_MODEL``
+~~~~~~~~~~~~~~
+
+If set, this should be the path to a model to look up allowed origins, in the
+form ``app.modelname``. Defaults to ``None``.
+
+The model should inherit from ``corsheaders.models.AbstractCorsModel`` and specify
+the allowed origin in the ``CharField`` called ``cors``.
+
+CSRF Integration
+----------------
+
+Most sites will need to take advantage of the `Cross-Site Request Forgery
+protection <https://docs.djangoproject.com/en/dev/ref/csrf/>`_ that Django
+offers. CORS and CSRF are distinct concepts, and Django will not use your CORS
+configuration to exempt sites from the ``Referer`` checking that it does on
+secure requests. The way to do that is with the `CSRF_TRUSTED_ORIGINS setting
+<https://docs.djangoproject.com/en/dev/ref/settings/#csrf-trusted-origins>`_.
+For example:
+
+.. code-block:: python
+
+    CORS_ORIGIN_WHITELIST = (
+        'read.only.com',
+        'change.allowed.com',
+    )
+    CSRF_TRUSTED_ORIGINS = (
+        'change.allowed.com',
+    )
+
 ``CORS_REPLACE_HTTPS_REFERER``
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-If ``True``, the ``HTTP_REFERER`` header will get replaced when CORS checks
-pass, so that the Django CSRF middleware checks work with HTTPS. Defaults to
+``CSRF_TRUSTED_ORIGINS`` was introduced in Django 1.9, so users of earlier
```

Comment:
I tentatively agree, though as a Warning rather than an Error.

I'm hesitant to "officially" deprecate this usage because the `CORS_REPLACE_HTTPS_REFERER` approach is more powerful than the `CSRF_TRUSTED_ORIGINS` approach; the latter is a hardcoded list of domains, whereas the former bypasses the `Referer` check for all CORS-approved sites, taking into account the regex and `*` settings.

Now, if those are real use cases that should be supported then the right place to do that is in Django, not here. But in the meantime, people may need to continue to use the `Referer`-forging approach to achieve their current or desired functionality.